hazelcast/src/main/java/com/hazelcast/nio/serialization/FieldType.java

```diff
@@ -55,4 +71,32 @@ public static FieldType get(byte type) {
         return ALL[type];
     }
 
+    public boolean isArrayType() {
+        return type >= PORTABLE_ARRAY.type;
```

Comment:
I was actually thinking of something like this, explicitly setting up the properties that describe the `FieldType`s and returning them from `isArrayType` and `getSingleType`. Apart from being at least as fast, I think it also models more explicitly what `FieldType` is about (where being explicit can be considered either good, or bad :) ).
```java
public enum FieldType {
    // SINGLE-VALUE TYPES
    PORTABLE(0, MAX_VALUE, 0, false),
    BYTE(1, BYTE_SIZE_IN_BYTES, 1, false),
....
    // ARRAY TYPES
    PORTABLE_ARRAY(10, MAX_VALUE, 0, true),
    BYTE_ARRAY(11, MAX_VALUE, 1, true),
.....
     private static final FieldType[] ALL = FieldType.values();

    private final byte type;
    private final byte elementType;
    private final int elementSize;
    private final boolean arrayType;

    FieldType(int type, int elementSize, int elementType, boolean arrayType) {
        this.type = (byte) type;
        this.elementSize = elementSize;
        this.elementType = (byte) elementType;
        this.arrayType = arrayType;
    }

    public byte getId() {
        return type;
    }

    public static FieldType get(byte type) {
        return ALL[type];
    }

    public boolean isArrayType() {
        return arrayType;
    }

    public FieldType getSingleType() {
        if (arrayType) {
            return get(elementType);
        }
        return this;
    }
```